Sources/Persistence/PersistencePerformanceMetricsTracker.swift

```diff
@@ -0,0 +1,159 @@
+import Foundation
+import Result
+
+public protocol PersistencePerformanceMetricsTracker: PerformanceMetricsTracker {
+
+    /// A closure to be used when stopping measuring memory reads/writes.
+    typealias MemoryAccessStopClosure<E: Error> = (_ result: Result<(blobSize: UInt64, memorySize: UInt64), E>) -> Void
+
+    /// A closure to be used when stopping measuring disk reads/writes.
+    typealias DiskAccessStopClosure<E: Error> = (_ result: Result<(blobSize: UInt64, diskSize: UInt64), E>) -> Void
+
+    /// The metadata key used for the used memory.
+    var usedMemoryMetadataKey: PerformanceMetrics.Metadata.Key { get }
```

Comment:
If you add a `typealias Metadata = PerformanceMetrics.Metadata` to `PerformanceMetricsTracker`, you don't need to keep repeating so much code, making it easier to read.

Reply:
Well spotted. We already have those `typealias`'es on `PerformanceMetricsTracker`, forgot to clean these up here. Will fix 😄